FirebaseAuth/Sources/Swift/Backend/RPC/AuthMFAResponse.swift

```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Protocol for responses that support Multi-Factor Authentication.
+protocol AuthMFAResponse {
+  /// An opaque string that functions as proof that the user has successfully passed the first
+  /// factor check.
+  func mfaPendingCredential() -> String?
+
+  /// Info on which multi-factor authentication providers are enabled.
+  func mfaInfo() -> [AuthProtoMFAEnrollment]?
```

Comment:
This might be a small improvement: to simplify the work that `EmailLinkSignInResponse`, etc. has to do to conform to this protocol, I believe you could instead define these methods as get-only properties. Gettable-only property requirements can be fulfilled with a stored property (as opposed to a gettable _and_ settable property requirement defined within a protocol).

```suggestion
  var mfaPendingCredential: String? { get }

  /// Info on which multi-factor authentication providers are enabled.
  var mfaInfo: [AuthProtoMFAEnrollment]? { get }
```

Reply:
Great suggestion! Done in the next commit.